src/main/java/com/marklogic/kafka/connect/sink/DefaultSinkRecordConverter.java

```diff
@@ -73,7 +105,29 @@ protected AbstractWriteHandle toContent(SinkRecord record) {
 			throw new NullPointerException("'record' must not be null, and must have a value.");
 		}
 		Object value = record.value();
+		Schema schema = record.valueSchema();
+		/* Determine the converter */
+		if (schema != null && value instanceof Struct) {
```

Comment:
I don't understand what's going on here. There doesn't seem to be a need to determine the converter, because "converter" is never used anywhere.

Reply:
This will be addressed in the follow-up refactor issue.